aioesphomeapi/client.py

```diff
@@ -693,15 +696,24 @@ def predicate_func(msg: BluetoothDeviceClearCacheResponse) -> bool:
 
         return BluetoothDeviceClearCache.from_pb(response)
 
-    async def bluetooth_device_disconnect(self, address: int) -> None:
+    async def bluetooth_device_disconnect(
+        self, address: int, timeout: float = DEFAULT_BLE_DISCONNECT_TIMEOUT
+    ) -> None:
         self._check_authenticated()
 
+        def predicate_func(msg: BluetoothDeviceConnectionResponse) -> bool:
+            return bool(msg.address == address and not msg.connected)
+
         assert self._connection is not None
-        self._connection.send_message(
+        await self._connection.send_message_await_response_complex(
```

Comment:
Waiting is fine because we always reply if already disconnected

https://github.com/esphome/esphome/blob/3498aade85f10e342efcd03bccc1dcdeceff5b5f/esphome/components/bluetooth_proxy/bluetooth_proxy.cpp#L272